src/prepare.c

```diff
@@ -576,13 +588,20 @@ static inline void pthreads_prepare_closures(pthreads_object_t *thread) {
 /* {{{ */
 static inline void pthreads_prepare_classes(pthreads_object_t* thread) {
 	zend_class_entry *entry;
+	zend_class_entry *prepared;
 	zend_string *name;
-
+	
 	ZEND_HASH_FOREACH_STR_KEY_PTR(PTHREADS_CG(thread->creator.ls, class_table), name, entry) {
 		if (!zend_hash_exists(PTHREADS_CG(thread->local.ls, class_table), name)) {
-			pthreads_prepared_entry(thread, entry);
+			pthreads_prepared_entry_internal(thread, entry, 0);
+		}
+	} ZEND_HASH_FOREACH_END();
+
+	ZEND_HASH_FOREACH_STR_KEY_PTR(PTHREADS_CG(thread->creator.ls, class_table), name, entry) {
+		if ((prepared = zend_hash_find_ptr(PTHREADS_CG(thread->local.ls, class_table), name)) && prepared->type != ZEND_INTERNAL_CLASS) {
+			pthreads_prepared_entry_static_members(thread, entry, prepared);
 		}
-	} ZEND_HASH_FOREACH_END(); 
+	} ZEND_HASH_FOREACH_END();
```

Comment:
This is a bit of a nitpick, but if you inverse the loops, then you can skip a check on each iteration:
```c
ZEND_HASH_FOREACH_STR_KEY_PTR(PTHREADS_CG(thread->local.ls, class_table), name, entry) {
    if (entry->type != ZEND_INTERNAL_CLASS) {
        pthreads_prepared_entry_static_members(
            thread,
            zend_hash_find_ptr(PTHREADS_CG(thread->creator.ls, class_table), name),
            entry
        );
    }
} ZEND_HASH_FOREACH_END();
```

I cannot immediately see anything wrong with this, since a child thread will only ever have classes that exist in the parent thread.